src/main/java/io/confluent/idesidecar/restapi/util/RetryableExecutor.java

```diff
@@ -0,0 +1,158 @@
+package io.confluent.idesidecar.restapi.util;
+
+import io.quarkus.logging.Log;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.function.Predicate;
+
+/**
+ * Utility class to execute a retryable operation with a configurable number of retries and backoff
+ * strategy.
+ */
+public class RetryableExecutor {
```

Comment:
Did you look into using Mutiny's support for [retrying on failures](https://smallrye.io/smallrye-mutiny/2.5.1/tutorials/retrying/) instead of implementing the `RetryableExecutor` from scratch?

Reply:
Thanks for suggesting! I'd started looking into it but didn't fully go through with it. I've replaced the retry logic with Mutiny APIs instead.